samples/boards/nrf52/mesh/onoff-app/src/main.c

```diff
@@ -0,0 +1,694 @@
+		local_trans_id = net_buf_simple_pull_u8(buf);
+		printk("%s addr: 0x%02x state: 0x%02x trans_id:"
+				" 0x%02x src: 0x%04x\n",
+				__func__, model->elem->addr,
+				onoff_state->current,
+				local_trans_id, ctx->addr);
+	}
+
+	/* Pin set low turns on LED's on the nrf52840-pca10056 board */
+	gpio_pin_write(onoff_state->led_device,
+			onoff_state->led_gpio_pin,
+			onoff_state->current ? 0 : 1);
+
+	/* Only publish if there is an assigned address */
+	if (model->pub->addr == BT_MESH_ADDR_UNASSIGNED) {
+		return;
+	}
+
+	/*
+	 * If a server has a publish address, it is required to
+	 * publish status on a state change
+	 *
+	 * See Mesh Profile Specification 3.7.6.1.2
+	 */
+
+	if (onoff_state->previous != onoff_state->current) {
+		printk("%s publish last: 0x%02x cur: 0x%02x\n",
+				__func__, onoff_state->previous,
+				onoff_state->current);
+		onoff_state->previous = onoff_state->current;
+		bt_mesh_model_msg_init(msg,
+				BT_MESH_MODEL_OP_GEN_ONOFF_STATUS);
+		net_buf_simple_add_u8(msg, onoff_state->current);
+		err = bt_mesh_model_publish(model);
+		if (err) {
+			printk("%s bt_mesh_model_publish: err: %d\n",
+					__func__, err);
+		}
+	}
+}
+
+static void gen_onoff_set(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	printk("%s\n", __func__);
+
+	gen_onoff_set_unack(model, ctx, buf);
+	gen_onoff_get(model, ctx, buf);
+}
+
+static void gen_onoff_status(struct bt_mesh_model *model,
+			     struct bt_mesh_msg_ctx *ctx,
+			     struct net_buf_simple *buf)
+{
+	u8_t	state;
+
+	state = net_buf_simple_pull_u8(buf);
+
+	printk("Node 0x%04x received OnOff status from 0x%04x with state 0x%02x\n",
+			model->elem->addr, ctx->addr, state);
+}
+
+
+
+static int output_number(bt_mesh_output_action_t action, uint32_t number)
+{
+	printk("OOB Number: %u\n", number);
+
+	return 0;
+}
+
+static int output_string(const char *str)
+{
+	printk("OOB String: %s\n", str);
+	return 0;
+}
+
+static void prov_complete(u16_t net_idx, u16_t addr)
+{
+	printk("provisioning complete for net_idx: 0x%04x addr: 0x%04x\n",
+			net_idx, addr);
+	primary_addr = addr;
+	primary_net_idx = net_idx;
+}
+
+static void prov_reset(void)
+{
+	bt_mesh_prov_enable(BT_MESH_PROV_ADV | BT_MESH_PROV_GATT);
+}
+
+static u8_t dev_uuid[16] = { 0xdd, 0xdd };
+
+/*
+ * Button Pressed Interrupt Handler
+ */
+
+#define BUTTON_DEBOUNCE_DELAY_MS 250
+
+static void button_pressed(struct device *dev, struct gpio_callback *cb,
+		 uint32_t pins)
+{
+	time = k_uptime_get_32();
+
+	/* debounce the switch */
+	if (time < last_time + BUTTON_DEBOUNCE_DELAY_MS) {
+		return;
+	}
+
+	last_time = time;
+	sw.pin = pins;
+	k_work_submit(&sw.button_work);
+}
+
+/*
+ * Map GPIO pins to button number
+ */
+
+static uint8_t pin_to_sw(uint32_t pin)
+{
+
+	switch (pin) {
+	case BIT(SW0_GPIO_PIN): return 0;
+	case BIT(SW1_GPIO_PIN): return 1;
+	case BIT(SW2_GPIO_PIN): return 2;
+	case BIT(SW3_GPIO_PIN): return 3;
+	}
+	printk("no match for pin\n");
```

Comment:
Minor thing: add an empty line after }. That's the general coding style we try to follow. The only exception is when the next line is also a }, in which case there's no empty line between them. You might also want to include the unmatched pin number in the log message.

Reply:
ok